compiler/modules/front/expr.bal

```diff
@@ -1141,12 +1149,12 @@ function codeGenFunctionCallExpr(ExprContext cx, bir:BasicBlock bb, s:FunctionCa
 function codeGenMethodCallExpr(ExprContext cx, bir:BasicBlock bb, s:MethodCallExpr expr) returns CodeGenError|ExprEffect {
     var { result: target, block: curBlock } = check codeGenExpr(cx, bb, (), expr.target);
     bir:FunctionRef func = check getLangLibFunctionRef(cx, target, expr.methodName, expr.namePos);
-    check validArgumentCount(cx, func, expr.args.length() + 1, expr.opPos);
+    check validArgumentCount(cx, func, expr);
 
     t:SemType[] paramTypes = func.signature.paramTypes;
     bir:Operand[] args = [target];
     foreach int i in 0 ..< expr.args.length() {
-        var { result: arg, block: nextBlock } = check codeGenExpr(cx, curBlock, paramTypes[i + 1], expr.args[i]);
+        var { result: arg, block: nextBlock } = check codeGenExprForType(cx, curBlock, expr.args[i], paramTypes[i + 1]);
         curBlock = nextBlock;
         args.push(arg);
     }
```

Comment:
Add this one
```
function codeGenArgument(ExprContext cx, bir:BasicBlock nextBlock, bir:MethodCallExpr|s:FunctionCallExpr callExpr, bir:FunctionRef func, int i) returns ExprEffect|CodeGenError { }
```

Have error saying "incorrect type for argument".
Get rid of codeGenForType.
Handle too many arguments here.

Then do `check sufficientArguments(cx, func, expr);` at the end.